lib/Data/Validate/Sanctions.pm

```diff
@@ -316,16 +318,20 @@ sub get_sanctioned_info {    ## no critic (RequireArgUnpacking)
 sub _load_data {
     my $self          = shift;
     my $sanction_file = $self->{sanction_file};
-    $self->{last_time}               //= 0;
+    $self->{last_modification}       //= 0;
+    $self->{last_index}              //= 0;
     $self->{_data}                   //= {};
     $self->{_sanctioned_name_tokens} //= {};
     $self->{_token_sanctioned_names} //= {};
 
     if (-e $sanction_file) {
-        return $self->{_data} if stat($sanction_file)->mtime <= $self->{last_time} && $self->{_data};
-        $self->{last_time} = stat($sanction_file)->mtime;
-        $self->{_data}     = LoadFile($sanction_file);
+        return $self->{_data} if stat($sanction_file)->mtime <= $self->{last_modification} && $self->{_data};
+        $self->{last_modification} = stat($sanction_file)->mtime;
```

Comment:
the `stat` requires reading from the file system and it's not a free operation, I'd set some threshold when we even don't bother to check modification time. for example, if less than  5-10 min passed from the last  check, do nothing.